types/turing-yttc-generated.yaml

```diff
@@ -0,0 +1,183 @@
+            name="tm:initialize"><optional><element name="tm:tape-content" nma:default=""><a:documentation>The
+            string with which the tape shall be initialized. The
+
+            leftmost symbol will be at tape coordinate 0.</a:documentation><data type="string"/></element></optional></element></nma:input></nma:rpc><nma:rpc><nma:input><element
+            name="tm:run"><empty/></element></nma:input></nma:rpc></nma:rpcs><nma:notifications><nma:notification><element
+            name="tm:halted"><a:documentation>The Turing Machine has halted. This
+            means that there is no
+
+            transition rule for the current state and tape symbol.</a:documentation><element
+            name="tm:state"><a:documentation>The state of the control unit in which
+            the machine has
+
+            halted.</a:documentation><ref name="turing-machine__state-index"/></element></element></nma:notification></nma:notifications></start></grammar></start><define
+            name="turing-machine__state-index"><a:documentation>Type for indexing
+            states of the control unit.</a:documentation><data type="unsignedShort"/></define><define
+            name="turing-machine__head-dir" nma:default="right"><a:documentation>Possible
+            directions for moving the read/write head, one cell
+
+            to the left or right (default).</a:documentation><choice><value>left</value><value>right</value></choice></define><define
+            name="turing-machine__cell-index"><a:documentation>Type for indexing tape
+            cells.</a:documentation><data type="long"/></define><define name="turing-machine__tape-symbol"><a:documentation>Type
+            of symbols appearing in tape cells.
+
+
+            A blank is represented as an empty string where necessary.</a:documentation><data
+            type="string"><param name="minLength">0</param><param name="maxLength">1</param></data></define><define
+            name="turing-machine___tape-cells"><a:documentation>The tape of the Turing
+            Machine is represented as a sparse
+
+            array.</a:documentation><zeroOrMore><element nma:key="coord" name="cell"><a:documentation>List
+            of non-blank cells.</a:documentation><element name="coord"><a:documentation>Coordinate
+            (index) of the tape cell.</a:documentation><ref name="turing-machine__cell-index"/></element><optional><element
+            name="symbol"><a:documentation>Symbol appearing in the tape cell.
+
+
+            Blank (empty string) is not allowed here because the
+
+            ''cell'' list only contains non-blank cells.</a:documentation><data type="string"><param
+            name="length">1</param></data></element></optional></element></zeroOrMore></define></grammar>'
+          modules:
+            turing-machine: '2013-12-27'
+          xmlns:
+            _: http://example.net/turing-machine
+      rpc:
+        required: false
+        type: _rpc
+
+#module: turing-machine
```

Comment:
Looks like it depended to real name of functionality/property on device - so we can change it only partially in yttc tool in case when we dont't use this "name" in communication with device, only as internal name for types.